### DISCUSSION TITLE
*Clickable* hot corners!

### DISCUSSION THREAD

**Discussion by navid-zamani** [linuxmint/discussions#96] [Ideas]
Back in the days, Compiz had hot corners too (and hot edges), but they had the option to only activate on a *click*. (And different actions for different mouse buttons too.)   
I loved them.

But nowadays, everyone’s only implementing auto-activating hot corners. Which are incredibly annoying: The mouse just always slips into these hot corners, and triggers the action inadvertently. Plus, the corners are already used by the launcher button, the close button of the fullscreen window, and the date display of the clock applet in the task bar. Making the desktop almost unusable until the hot corners are disabled. 

**So it would be wonderful, if the hot corners settings got not just one drop-down for actions, but one for each mouse button** (up to 8 or so) and one for no mouse button.  
Especially given how easy it is to implement this. (I wrote an entire hot corner application in two hours. Three if you count having to learn X window management first. But it will always be inferior due to not being part of the window manager itself.)

Plus, the prerequisite of having to click too, makes hot edges unproblematic and usable too, without conflicting with anything or being triggered too often. Especially since nobody would configure clicks that would conflict with existing clickable areas there.

I used to set up my entire window management with them. One corner for the launcher [(M) button], the opposing one for closing the current window, one for the task switcher, and one for my personal planning tool. Edge clicks did switch between desktops. (I’m non using floating windows and consider them a marketing-originated misfeature, so all my windows are full-screen or tiled. Plus every window had a fixed place. So I didn’t need to resize or move windows.)

What do you think?

**Comment by Hilyxx** [comment]
There is a button for expo or scale mod in "Cornerbar" applet. you need to configure applet settings.. that's all

**Reply by navid-zamani** [reply]
That requires an actual bar to be there to put the applet in. Unlike the built-in thing.

Also, it is strange, that that isn’t just built-in. I mean there is zero sane use case for the infuriating no-click one. Everyone who tries it, immediately disables it again, the first time they shove their mouse aside and their windows fly all over the place when they wanted to type something. Merging the cornerbar code to the built-in thing or vice vera, would avoid code duplication and get us at least one usable UI element.

Can you point me to the code for the built-in thing? (Including the settings dialog.) Can’t be more than five minutes to do that.

**Comment by navid-zamani** [comment]
Ok, found it and … done. It now only activates on clicks. I threw the entire on-enter and delay stuff out.

But I don’t know how to handle different mouse buttons (and modifier keys held) yet. I don’t know what APIs these are. Is it GTK? `button-release-event` is a GTK function, it seems… but it takes a parameter, and here there is no parameter in the `actor.connect(…)` line nor in `_onCornerClicked()`. Also I don‘t know how to debug it. (E.g. step through, trace, call stack, etc.)
